app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/weapons/list/WeaponTreeListAdapter.kt

```diff
@@ -81,6 +75,57 @@ class WeaponTreeListAdapterDelegate(private val onSelected: (WeaponTree) -> Unit
             view.slot1.setImageDrawable(slotImages[0])
             view.slot2.setImageDrawable(slotImages[1])
             view.slot3.setImageDrawable(slotImages[2])
+
+            // DYNAMIC STATS
+
+            // Clear the placeholder layouts
+            view.complex_stat_layout.removeAllViews()
+
+            // Elemental Stat
+            if (weaponTree.element1 != null){
+                val elementView = CompactStatCell(
+                        view.context,
+                        getElementIcon(view.context, weaponTree.element1),
+                        createElementString(weaponTree.element1_attack, weaponTree.element_hidden))
+
+                if (weaponTree.element_hidden) {
+                    elementView.labelView.alpha = 0.5.toFloat()
+                } else {
+                    elementView.labelView.alpha = 1.0.toFloat()
+                }
+
+                view.complex_stat_layout.addView(elementView)
+            }
+
+            if (weaponTree.affinity != 0) {
+                val affinitySb = StringBuilder()
+                val prepend = if (weaponTree.affinity > 0) "+" else ""
+                affinitySb.append(prepend).append(weaponTree.affinity).append("%")
```

Comment:
I think its cleaner to use an interpolated string. StringBuilder requires a double take when reading.